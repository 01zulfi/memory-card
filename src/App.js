import React from "react";
import Header from "./components/Header";
import Scoreboard from "./components/Scoreboard";

function App() {
  return (
    <div className="App">
      <Header />
      <Scoreboard currentScore={null} bestScore={null} />
    </div>
  );
}

export default App;
