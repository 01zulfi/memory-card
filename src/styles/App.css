div.App {
  display: flex;
  width: 100%;
  height: 100vh;
  align-items: center;
  flex-direction: column;
  background-color: rgb(182, 214, 204, 0.2);
}
