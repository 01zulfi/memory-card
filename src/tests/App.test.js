import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import App from "../App";

test("App renders", () => {
  const { container } = render(<App />);

  screen.debug();

  expect(container).toMatchSnapshot();
  expect(container.firstChild).toHaveClass("App");
});

test("header renders", () => {
  const { container } = render(<App />);
  const header = container.querySelector(".header");
  expect(header).toBeInTheDocument();
});

test("scoreboard renders", () => {
  const { container } = render(<App />);
  const scoreboard = container.querySelector(".scoreboard");
  expect(scoreboard).toBeInTheDocument();
});

test("ball section renders", () => {
  const { container } = render(<App />);
  const ballSection = container.querySelector(".ball-section");
  expect(ballSection).toBeInTheDocument();
});
