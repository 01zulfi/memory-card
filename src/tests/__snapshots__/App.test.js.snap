// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="header"
    >
      <h1
        class="title"
      >
        Memory Card
      </h1>
    </div>
    <div
      class="scoreboard"
    >
      <div
        class="current-score-div"
      >
        <p>
          Current Score
        </p>
        <p />
      </div>
      <div
        class="best-score-div"
      >
        <p>
          Best Score
        </p>
        <p />
      </div>
    </div>
    <section
      class="ball-section"
    >
      <div
        class="ball-div"
        data-id="1114007277401"
      >
        Football
      </div>
      <div
        class="ball-div"
        data-id="1236213125317"
      >
        Squash Ball
      </div>
      <div
        class="ball-div"
        data-id="111269804956"
      >
        Golf Ball
      </div>
      <div
        class="ball-div"
        data-id="1306394998416"
      >
        Basketball
      </div>
    </section>
  </div>
</div>
`;
